src/keys.rs

```diff
@@ -256,3 +256,63 @@ impl<C: CipherSuite> GroupVerifyingKey<C> {
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::testing::Secp256k1Sha256;
+    use crate::utils::Scalar;
+    use crate::{FromBytes, ToBytes};
+    use ark_ff::UniformRand;
+    use rand::{rngs::OsRng, RngCore};
+
+    #[test]
+    fn test_serialization() {
+        let mut rng = OsRng;
+
+        for _ in 0..100 {
+            let skey = DiffieHellmanPrivateKey(Scalar::<Secp256k1Sha256>::rand(&mut rng));
+            let bytes = skey.to_bytes().unwrap();
+            assert!(DiffieHellmanPrivateKey::<Secp256k1Sha256>::from_bytes(&bytes).is_ok());
+            assert_eq!(
+                skey,
+                DiffieHellmanPrivateKey::<Secp256k1Sha256>::from_bytes(&bytes).unwrap()
+            );
+
+            let pkey = DiffieHellmanPublicKey::<Secp256k1Sha256>::new(
+                <Secp256k1Sha256 as CipherSuite>::G::generator().mul(skey.0),
+            );
+            let bytes = pkey.to_bytes().unwrap();
+            assert!(DiffieHellmanPublicKey::<Secp256k1Sha256>::from_bytes(&bytes).is_ok());
+            assert_eq!(
+                pkey,
+                DiffieHellmanPublicKey::<Secp256k1Sha256>::from_bytes(&bytes).unwrap()
+            );
+
+            let skey = IndividualSigningKey::<Secp256k1Sha256> {
+                index: rng.next_u32(),
+                key: Scalar::<Secp256k1Sha256>::rand(&mut rng),
+            };
+            let bytes = skey.to_bytes().unwrap();
+            assert!(IndividualSigningKey::<Secp256k1Sha256>::from_bytes(&bytes).is_ok());
+            assert_eq!(
+                skey,
+                IndividualSigningKey::<Secp256k1Sha256>::from_bytes(&bytes).unwrap()
+            );
+
+            let pkey = skey.to_public();
+            let bytes = pkey.to_bytes().unwrap();
+            assert!(IndividualVerifyingKey::<Secp256k1Sha256>::from_bytes(&bytes).is_ok());
+            assert_eq!(
+                pkey,
+                IndividualVerifyingKey::<Secp256k1Sha256>::from_bytes(&bytes).unwrap()
+            );
+        }
+
+        let bytes = vec![255; 157];
```

Comment:
Maybe it's obvious why this is an invalid key, but I don't get it. Either way: what do you think about moving this to its own test?

Reply:
I could add a comment on it, yes. The idea is that `[255; 157]` is just way too long (157 bytes, for a scalar field that is 32 bytes). (not sure where I picked this constant actually)

Reply:
aha, obviously! :)

Let's just rename the lvar from `bytes` to `too_long_key` or something?

Reply:
I still think it should be its own test though.